app/controllers/db_files_controller.rb

```diff
@@ -0,0 +1,12 @@
+class DbFilesController < ApplicationController
+  skip_before_action :check_admin
+
+  def show
+    # TODO: Generate an external key instead of using internal id
+    # because we have no access control on a file
```

Comment:
We can lock down all files, except files submitted as part of solutions. i.e. if a student wants to download their submission, we can validate that the file belongs to a submission that belongs to them. Otherwise, reject.